src/fmu/dataio/_model/schema.py

```diff
@@ -134,18 +129,18 @@ def _deprecation_warning(self) -> UnsetContent:
         return self
 
 
-class UnsetAnyContent(data.AnyData):
-    root: UnsetContent  # type: ignore
+class InternalAnyData(data.AnyData):
+    root: InternalUnsetData  # type: ignore
 
 
-class FMUClassMetaData(fields.FMU):
+class InternalFMU(fields.FMU):
     # This class is identical to the one used in the schema
     # exept for more fmu context values beeing allowed internally
     context: Context  # type: ignore
 
 
-class DataClassMeta(JsonSchemaMetadata):
-    # TODO: aim to use meta.FMUDataClassMeta as base
+class InternalObjectMetadata(JsonSchemaMetadata):
+    # TODO: aim to use InternalFMU as base
```

Comment:
I think this comment adjustment is not quite right, but is indicating that we should try not to derive from what is now `ObjectMetadata` in the main Pydantic model. `JsonSchemaMetadata` is essentially a hack to add `$schema:"https://json-schema.org/draft/2020-12/schema"` to the resulting schema so that it validates correctly against Sumo's schema validation library. This is because Pydantic generates an OpenAPI schema which does not by default contain this `$schema` attribute.

I definitely understand the confusion between `FMUDataClassMeta` and `FMUClassMetaData` though 😄